descartes_planner/src/planning_graph.cpp

```diff
@@ -763,95 +761,53 @@ void PlanningGraph::printGraph()
   ROS_DEBUG_STREAM("\n\nEND PRINTING GRAPH\n\n");
 }
 
-bool PlanningGraph::calculateJointSolutions()
+bool PlanningGraph::calculateJointSolutions(const std::vector<TrajectoryPtPtr>& points,
+                                            std::vector<std::vector<JointTrajectoryPt>>& poses)
 {
-  if (joint_solutions_map_.size() > 0)
-  {
-    // existing joint solutions... clear the list?
-    ROS_WARN_STREAM("existing joint solutions found, clearing map");
-    joint_solutions_map_.clear();
-  }
+  poses.resize(points.size());
 
-  // for each TrajectoryPt, get the available joint solutions
-  for (auto trajectory_iter = cartesian_point_link_->begin();
-       trajectory_iter != cartesian_point_link_->end(); ++trajectory_iter)
+  for (std::size_t i = 0; i < points.size(); ++i)
   {
-    // TODO: copy this block to a function that can be used by add and modify
-    /*************************/
-    std::vector<TrajectoryPt::ID> traj_solutions;
-    std::vector<std::vector<double> > joint_poses;
-    trajectory_iter->second.source_trajectory_.get()->getJointPoses(*robot_model_, joint_poses);
-    TrajectoryPt::ID tempID = trajectory_iter->first;
-    ROS_DEBUG_STREAM("CartID: " << tempID << " JointPoses count: " << joint_poses.size());
+    std::vector<std::vector<double>> joint_poses;
+    points[i]->getJointPoses(*robot_model_, joint_poses);
 
-    if (joint_poses.size() == 0)
+    if (joint_poses.empty())
     {
-      ROS_WARN_STREAM("no joint solution for this point... potential discontinuity in the graph");
+      ROS_ERROR_STREAM(__FUNCTION__ << ": IK failed for input trajectory point with ID = " << points[i]->getID());
+      return false;
```

Comment:
I'm surprised this wasn't returning **false** already.  It feels to me as that is the correct way to handle this when there are no joint solution; however were we deliberately ignoring this before?